### ISSUE TITLE
Copy labels from admin instead of project

### ISSUE THREAD

**Issue by mackeyja92** [gotsunami/gitlab-copy#18]
I want to define labels at the site level and just copy to each project. How can I copy labels without specifying a project?

**Comment by matm** [comment]
@mackeyja92 What do you mean labels at site level? Aren't labels tied to projects? I can't find a link about global label definition.

There's no way to copy all labels from a source project to all other projects at the moment. There may be a need for that feature indeed.

**Comment by Hufschmidt** [comment]
I think he means group labels which can be created by going to `[some group or sub-group] > issues > label`. Group labels can be used in any project in this group but not any sub-group or sub-group project.
Here is where gitlab-copy would come in handy.

**Comment by matm** [comment]
Yes, indeed. I wasn't aware of those. [some group] > edit > labels allows to define labels for all projects in a group. Thanks @Hufschmidt  for pointing this out. How do you think `gitlab-copy` could handle it? Allow copying labels from one project to a group's labels? Maybe also allow copying of labels between groups.

**Comment by Hufschmidt** [comment]
Since gitlab labels of a group are currently only available in projects that are in that group, but not in projects in any of its sub-groups ([Gitlab issue](https://gitlab.com/gitlab-org/gitlab-ce/issues/2651)) my use-case would be copying labels from one group to another.

Side-note: As far as I know project-labels can already be *promoted* to group-labels via gitlabs own gui, having this in gitlab-copy might still be usefull when trying to migrate a lot of labels from project to group though.

I have to admin though I don't know if there is an API endpoint for fetching or adding group-level labels, last time I looked I didn't find any documentation on this...

**Comment by matm** [comment]
Hey there! Is copying labels from one group to another still a wanted feature? Just want to know if I work on this (to be included to upcoming 0.7 release). Thanks!

**Comment by zaventh** [comment]
Copying labels from group to group even in a single instance would be very desirable @matm. Currently I have a crude script to do this with `curl` and `jq`.
